### PR TITLE
Tasks

### PR DESCRIPTION
**Pull request by omgaikwad**
features added:
API Calls to:
- Add a Task
- Delete a Task
- Edit a Task
- Completed a Task

### CODE REVIEW

**Review by prathmeshjagtap — APPROVED** [review]
The code is Clean and functionality is also working properly

File: src/context/task-context.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        "/api/habits",
+        { habit: taskObj },
+        {
+          headers: {
+            authorization: auth.token,
+          },
+        }
+      );
+      setTasks(response.data.habits);
+      console.log(response.data.habits);
```
Comment (narharikale) [inline src/context/task-context.js]: console.log here

File: src/context/task-context.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        `/api/habits/${taskId}`,
+        { habit: taskObj },
+        {
+          headers: {
+            authorization: auth.token,
+          },
+        }
+      );
+      setTasks(response.data.habits);
+      console.log(response.data.habits);
```
Comment (narharikale) [inline src/context/task-context.js]: here also

**Review by narharikale — APPROVED** [review]
good to go 🚀

File: src/context/task-context.js (hunk 1 above)
Comment (t007rushi) [reply src/context/task-context.js]: rightly pointed out by @narharikale

File: src/context/task-context.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        setTasks(taskResponse.data.habits);
+      })();
+    } else {
+      setTasks([]);
+    }
+  }, [auth]);
+
+  const getSingleTask = async (taskId) => {
+    try {
+      const response = await axios.get(`/api/habits/${taskId}`, {
```
Comment (t007rushi) [inline src/context/task-context.js:30]: destructure here itself ```{data}``` makes it more readable

File: src/pages/LandingPage/LandingPage.jsx [hunk 4]
```diff
@@ -19,11 +27,13 @@ const LandingPage = () => {
...
               Track Time.
             </h1>
-            <Link to="/tasks">
-              <button className="bg-serene-red text-white font-bold px-12 py-4 rounded-3xl rounded-br-none mt-16 ml-12">
-                Get Started
-              </button>
-            </Link>
+
+            <button
+              onClick={() => getStartedBtnHandler()}
```
Comment (t007rushi) [inline src/pages/LandingPage/LandingPage.jsx:32]: simply put function name no need of arrow function here
```{getStartedBtnHandler}``` works as callback

File: src/pages/TaskPage/TaskPage.jsx [hunk 5]
```diff
@@ -1,93 +1,224 @@
...
-                </button>
-                <button className="text-2xl text-serene-red-light">
-                  <AiFillDelete />
-                </button>
-              </div>
-            </div>
+          <div className="task-list flex flex-col gap-6 items-center">
+            {tasks.map((task) => {
+              return (
+                <div className="single-task-container flex items-center border-dashed border-2 border-serene-red-light py-4 px-8 rounded-lg cursor-pointer w-4/5">
```
Comment (t007rushi) [inline src/pages/TaskPage/TaskPage.jsx:174]: map is used Add ```key``` prop here

**Review by t007rushi — APPROVED** [review]
naming is quite good 
with some suggestions being done
Good to go ahead

**Pull request merged**